File: src/main/java/org/example/scalingstream/partitioner/HashPartitioner.java
package org.example.scalingstream.partitioner;

import org.example.scalingstream.Record;

public class HashPartitioner extends Partitioner {
    HashPartitioner(int numOut) {
        super(numOut);
    }

    @Override
    public int assignPartition(Record record) {
        return record.hashCode();
    }
}

File: src/main/java/page/parekh/scalingstream/partitioner/HashPartitioner.kt
package page.parekh.scalingstream.partitioner

import kotlin.math.abs


class HashPartitioner : Partitioner {
    override val name: String = "HASH"

    override fun <Type> assignPartition(record: Type, numOut: Int): Int {
        if (numOut == 0) error("Can't partition into zero partitions.")
        return abs(record.hashCode() % numOut)
    }
}
